### PR TITLE
Release 1.18.0

### PR DESCRIPTION
**Pull request by zmoog**
## What does this PR do?

Set up ESF release 1.18.0

## Checklist

- ~~[ ] My code follows the style guidelines of this project~~
- ~~[ ] I have commented my code, particularly in hard-to-understand areas~~
- ~~[ ] I have made corresponding changes to the documentation~~
- ~~[ ] I have made corresponding change to the default configuration files~~
- ~~[ ] I have added tests that prove my fix is effective or that my feature works~~
- [x] I have added an entry in `CHANGELOG.md` and updated `share/version.py`, if my change requires a new release.

### CODE REVIEW

**Review by constanca-m — DISMISSED** [review]

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,9 @@
+### v1.17.3 - 2024/10/08
+##### Bug fixes
+* Dead letter index (DLI): align the `error` field in documents to the ECS format; the field now provides `error.message` and `error.type` [793](https://github.com/elastic/elastic-serverless-forwarder/pull/793A).
```
Comment (constanca-m) [inline CHANGELOG.md:3]:
```suggestion
* Dead letter index (DLI): align the `error` field in documents to the ECS format; the field now provides `error.message` and `error.type` [793](https://github.com/elastic/elastic-serverless-forwarder/pull/793).
```

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,3 +1,9 @@
...
+##### Features
+* Dead letter index (DLI): do not send retryable errors to the DLI [793](https://github.com/elastic/elastic-serverless-forwarder/pull/793A).
```
Comment (constanca-m) [inline CHANGELOG.md:5]:
```suggestion
* Dead letter index (DLI): do not send retryable errors to the DLI [793](https://github.com/elastic/elastic-serverless-forwarder/pull/793).
```

Comment (zmoog) [reply CHANGELOG.md:5]: It's unbelivable how many errors I can make today in such a small PR.

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,3 +1,9 @@
+### v1.17.3 - 2024/10/08
```
Comment (kaiyan-sheng) [inline CHANGELOG.md]:
```suggestion
### v1.18.0 - 2024/10/08
```

**Review by kaiyan-sheng — APPROVED** [review]

**Pull request merged**